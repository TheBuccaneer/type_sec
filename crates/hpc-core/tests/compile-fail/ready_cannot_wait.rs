// Dieser Test soll NICHT kompilieren: Auf Ready gibt es kein `wait()`.
//
// trybuild erwartet, dass der Build fehlschlägt.
// Wir konstruieren absichtlich einen Ready-Buffer und rufen `wait()` darauf auf.

use hpc_core::Ready;

fn main() {
    // Wir brauchen keinen echten Buffer; es geht nur um den Typ.
    // SAFETY: absichtlich unsicher für Compile-Fail; es wird nicht ausgeführt.
    let buf: hpc_core::GpuBuffer<Ready> = unsafe { std::mem::zeroed() };

    // darf es auf `Ready` NICHT geben:
    let _bad = buf.wait();
}
