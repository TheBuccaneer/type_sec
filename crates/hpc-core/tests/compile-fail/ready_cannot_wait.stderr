error[E0599]: no method named `wait` found for struct `GpuBuffer` in the current scope
 --> tests/compile-fail/ready_cannot_wait.rs
  |
  |     let _bad = buf.wait();
  |                    ^^^^ method not found in `GpuBuffer<hpc_core::Ready>`
